test/e2e/scenarios/add_cluster.go

```diff
@@ -0,0 +1,62 @@
+package scenarios
+
+import (
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = PDescribe("[Dataplane] Adding existng network policy to newly added cluster", func() {
+	PContext("Registering cluster 3 with kubefed and then creating a pod", func() {
+		PIt("Should implement existng network policy in newly added cluster", func() {
+			By("creating listener pod 1 with label 1 in cluster 1 in namespace 1")
+			By("creating connecting pod 2 with label 2 in cluster 2 in namespace 2")
+			By("creating network policy with namespace selector on cluster 1")
+			By("testing connectivity between pods")
+			By("Registering cluster 3 to kubefed")
+			By("creating connecting pod 3 with label 3 in cluster 3 in namespace 3")
+			By("testing connectivity between pod 1 and 2")
+			By("testing connectivity between pod 1 and 3")
+			By("Unregistering cluster 3")
+		})
+	})
+
+	PContext("[Dataplane] Creating a pod in cluster 3 and then registering it with kubefed", func() {
+		PIt("Should implement existng network policy in newly added cluster", func() {
+			By("creating listener pod 1 with label 1 in cluster 1 in namespace 1")
+			By("creating connecting pod 2 with label 2 in cluster 2 in namespace 2")
+			By("creating network policy with namespace selector on cluster 1")
+			By("testing connectivity between pod 1 and 2")
+			By("creating connecting pod 3 with label 3 in cluster 3 in namespace 3")
+			By("Registering cluster 3 to kubefed")
+			By("testing connectivity between pod 1 and 3")
```

Comment:
You cannot re-use listener pods, listener pods go away as soon as one connector pod connects, you need two listener pods.

Reply:
Done